common/statsd/statsd.cc

```diff
@@ -73,7 +75,11 @@ class StatsdClientWrapper {
 
 public:
     StatsdClientWrapper(string host, int port, string prefix)
-        : link(statsd_init_with_namespace(host.c_str(), port, cleanMetricName(prefix).c_str())) {}
+        : link(statsd_init_with_namespace(host.c_str(), port, cleanMetricName(prefix).c_str())) {
+        if (link == nullptr) {
+            Exception::raise("statsd initialization failed: host: {} port: {} prefix: {}", host, port, prefix);
```

Comment:
slight preference for using `host={}` instead of `host: {}` as it makes it easier to have splunk ingest these lines

Reply:
done!